src/CodegenShapeFuture.php

```diff
@@ -0,0 +1,63 @@
+<?hh // strict
+/*
+ *  Copyright (c) 2015-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+namespace Facebook\HackCodegen;
+
+/**
+ * Generate code for a shape. Please don't use this class directly; instead use
+ * the function codegenShape_FUTURE.  E.g.:
+ *
+ * codegenShape_FUTURE(vec([
+ *   new CodegenShapeMember('x', 'int'),
+ *   new CodegenShapeMember('y', 'int'),
+ * ]))
+ */
+final class CodegenShapeFuture implements ICodeBuilderRenderer {
```

Comment:
Can you also rename this to the `_FUTURE` pattern, just to make clear it's not a future a.ka. promise :)